### ISSUE TITLE
Provide basic usage examples in the documentation

### ISSUE THREAD

**Issue by erikpols** [dimforge/parry#69]
As in title. I have the feeling you guys are seriously underestimating the importance of this. 

Examples required:
- how to do basic intersections between segments
- how to do ray casting
- how to get the distance between points
- how to use the polygon shape (figuring out whether it is closed loop, doing intersections etc)

Even if you just 'dump' say 10 code examples in the index of the documentation, that would already go a long way.

**Comment by EriKWDev** [comment]
I agree the documentation is very bare bones, but shout-out to the examples directory https://github.com/dimforge/parry/tree/master/crates/parry2d/examples as well as the tests at https://github.com/dimforge/parry/tree/master/crates/parry2d/tests

More complete examples would be nice. Perhaps more of a complete demo like a platformer using some rendering library like macroquad together with physics from parry would be greatly appreciated.
